deviceModeInit.js

```diff
@@ -130,26 +130,23 @@ var rudderTracking = (function () {
     } else {
       console.debug("No heap cookie found.");
     }
-
-    htmlSelector.buttonAddToCart =
-      rs$('form[action="/cart/add"] [type="submit"]')
+    htmlSelector.buttonAddToCart = rs$('form[action="/cart/add"] [type="submit"]')
     fetchCart()
       .then((cart) => {
         const needToUpdateCart = checkCartNeedsToBeUpdated(cart);
         if (userLoggedOut || needToUpdateCart) {
           updateCartAttribute().then((cart) => {
-            sendIdentifierToRudderWebhook(cart);
-            console.debug("Successfully updated cart");
+            sendIdentifierToRudderWebhook(cart); // sending rudderIdentifier periodically after this
+            sendSessionIdentifierToRudderWebhook(cart); // sending sessionIdentifier
+            console.log("Successfully updated cart");
+            checkAndSendSessionRudderIdentifierPeriodically();
+            checkAndSendRudderIdentifier(cart, delay = 10000);
```

Comment:
Calling the above functions in a way that they do not directly check the need to update the cart attributes as cookie update will take some time